Sources/StreamChatUI/Navigation/NavigationRouter.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// A root class for all routes in the SDK.
+///
+/// Router objects are used to handle navigation between view controllers.
+///
+/// - Important: ⚠️ The lifetime of a router must be similar to the lifetime of its `rootViewController`. Only the
+/// root view controller is allowed to hold a strong reference to its router object.
+///
+open class NavigationRouter<Controller: UIViewController>: UIResponder {
+    /// The root `UIViewController` object of this router.
+    public unowned var rootViewController: Controller
+
+    /// A convenience method to get the navigation controller of the root view controller.
+    public var rootNavigationController: UINavigationController? {
+        rootViewController.navigationController
+    }
+
+    override open var next: UIResponder? {
+        rootViewController.next
+    }
```

Comment:
booooo, that is very nice trick 👍

Reply:
Not sure how it will work IRL though :)

Reply:
What is the intention with this? It does not seem where are using it anywhere in the PR 🤔

Reply:
It's because we want to have access to `Components` and you need to be part of the responder chain for that. I'll added a comment 👍